polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,52 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
+  currency?: string;
+  timezone?: string;
+}
```

Comment:
The `i18next` provider doesn't store state for things other than locale, so I have to create a new context. This works well because we can pass these values into whatever other providers that a component might use